module-system/sov-modules-api/src/hooks.rs

```diff
@@ -28,25 +29,38 @@ pub trait TxHooks {
 }
 
 /// Hooks related to the Sequencer functionality.
-/// In essence, the sequencer locks a bond at the beginning of the `StateTransitionFunction::apply_blob`,
+/// In essence, the sequencer locks a bond at the beginning of the `StateTransitionFunction::apply_tx_blob`,
 /// and is rewarded once a blob of transactions is processed.
 pub trait ApplyBlobHooks {
     type Context: Context;
     type BlobResult;
 
-    /// Runs at the beginning of apply_blob, locks the sequencer bond.
+    /// Runs at the beginning of apply_tx_blob, locks the sequencer bond.
     /// If this hook returns Err, batch is not applied
     fn begin_blob_hook(
         &self,
         blob: &mut impl BlobReaderTrait,
         working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
     ) -> anyhow::Result<()>;
 
-    /// Executes at the end of apply_blob and rewards or slashed the sequencer
+    /// Executes at the end of apply_tx_blob and rewards or slashed the sequencer
     /// If this hook returns Err rollup panics
     fn end_blob_hook(
         &self,
         result: Self::BlobResult,
         working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
     ) -> anyhow::Result<()>;
 }
+
+/// Hooks that execute during the `StateTransitionFunction::begin_slot` and `end_slot` functions.
+pub trait SlotHooks<Cond> {
+    type Context: Context;
+
+    fn begin_slot_hook(
+        &self,
+        slot_data: &impl SlotData<Condition = Cond>,
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
+    ) -> anyhow::Result<()>;
+
+    fn end_slot_hook(&self, new_state_root: [u8; 32]) -> anyhow::Result<()>;
```

Comment:
What happens if end_slot_hook returns an error? 

I believe that introducing new hooks require alignment with https://github.com/Sovereign-Labs/sovereign-sdk/issues/434